src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -792,21 +792,36 @@ public override string ToString()
         {
             if (_sddlForm == null)
             {
-                StringBuilder result = new StringBuilder();
-
                 //
                 // Typecasting of _IdentifierAuthority to a long below is important, since
                 // otherwise you would see this: "S-1-NTAuthority-32-544"
                 //
-
-                result.Append("S-1-").Append((long)_identifierAuthority);
-
-                for (int i = 0; i < SubAuthorityCount; i++)
+                // length of buffer calculation
+                // prefix = "S-1-".Length: 4;
+                // authority: long.MaxValue.ToString("D") + '-'.Length : 18+1: 20;
+                // subauth = MaxSubAuthorities * ( uint.MaxValue.ToString("D").Length + '-'.Length ): 15 * (10+1): 165;
+                // max possible length = 4 + 20 + 165: 189;
+                Span<char> result = stackalloc char[189];
+                int written = 0;
+                int length = 4;
+                ReadOnlySpan<char> prefix = stackalloc char[] { 'S', '-', '1', '-' };
+                int count = _subAuthorities.Length;
+                int[] values = _subAuthorities;
+
+                prefix.CopyTo(result);
+                ((long)_identifierAuthority).TryFormat(result.Slice(4), out written);
+                length += written;
+
+                for (int index = 0; index < count; index++)
```

Comment:
It would be better for this to go to values.Length rather than to count.

Reply:
1% faster on direct comparison, noise compared to the original. Why is it better to use the length?

Reply:
If you use `values.Length` for the condition and `values[i]` to access the element inside the loop, the JIT will eliminate the bounds check.